musit-service/src/main/scala/no/uio/musit/security/AuthenticatedUser.scala

```diff
@@ -62,8 +63,9 @@ case class AuthenticatedUser(
       logger.debug(s"User with GodMode accessing system.")
       MusitSuccess(())
     } else if (isAuthorizedFor(museum)) {
-      val allowed = permissionsFor(museum).exists(_.priority >= lowest.priority)
-      if (allowed) MusitSuccess(())
+      val allowedForPermissions = permissionsFor(museum).exists(_.priority >= lowest.priority)
+      val allowedForModule = module.foldLeft(true)((_, m) => this.groups.exists(_.module == m))
```

Comment:
Can you rewrite this to use pattern matcher. Easer to see what you're trying to achieve then. :)